sql/dm-example-sqlite.sql

```diff
@@ -0,0 +1,148 @@
+  json_extract(c.json, '$.id') ResourceId
+FROM
+  Condition c
+  join json_each(json_extract(c.json, '$.clinicalStatus.coding')) clinical_status_coding on (
+    json_extract(clinical_status_coding.value, '$.code') = 'active'
+  )
+  join json_each(json_extract(c.json, '$.code.coding')) condition_coding on (
+    json_extract(condition_coding.value, '$.system') = 'http://snomed.info/sct'
+    and json_extract(condition_coding.value, '$.code') in (
+      '10725009', '10964002', '1201005', '123799005', '59621000')
+  )
+  OR (
+    json_extract(condition_coding.value, '$.system') = 'http://hl7.org/fhir/ValueSet/icd-9'
+    and json_extract(condition_coding.value, '$.code') in ('401.1', '401.9')
+  );
+
+create temporary table a1c_2 AS
+SELECT
+  json_extract(o.json, '$.subject.reference_id_aa') PatientId,
+  json_extract(o.json, '$.id') ResourceId,
+  json_extract(o.json, '$.valueQuantity') valueQuantity,
+  json_extract(o.json, '$.effectiveDateTime_aa.start') effectiveDateTime_start,
+  json_extract(o.json, '$.effectiveDateTime_aa.end') effectiveDateTime_end
+FROM
+  Observation o
+  join json_each(json_extract(o.json, '$.code.coding')) observation_coding on (
+    json_extract(observation_coding.value, '$.system') = 'http://loinc.org'
+    and json_extract(observation_coding.value, '$.code') = '4548-4'
+  )
+where
+  json_extract(valueQuantity, '$.code') = '%'
+  and json_extract(valueQuantity, '$.system') = 'http://unitsofmeasure.org'
+  and json_extract(o.json, '$.status') in ('final', 'amended', 'corrected');
+
+create temporary table a1c_2_latest AS
+SELECT
+  *
+from
+  (
+    select
+      a1c_2.*,
+      dense_rank() over (
+        partition by a1c_2.PatientId
+        order by
+          a1c_2.effectiveDateTime_end DESC
+      ) as rank
+    from
+      a1c_2
+  ) as q
+where
+  rank = 1;
+
+create temporary table a1c_2_elevated AS
+select
+  *
+from
+  a1c_2_latest
+WHERE
+  json_extract(valueQuantity, '$.value') > 6;
+
+create temporary table a1c_1 AS
+SELECT
+  json_extract(o.json, '$.subject.reference_id_aa') PatientId,
+  json_extract(o.json, '$.id') ResourceId,
+  json_extract(o.json, '$.valueQuantity') valueQuantity,
+  json_extract(o.json, '$.effectiveDateTime_aa.start') effectiveDateTime_start,
+  json_extract(o.json, '$.effectiveDateTime_aa.end') effectiveDateTime_end
+FROM
+  Observation o
+  join a1c_2_elevated on (
+    a1c_2_elevated.PatientId = json_extract(o.json, '$.subject.reference_id_aa')
+    AND DATETIME(
+      json_extract(o.json, '$.effectiveDateTime_aa.end')
+    ) < DATETIME(a1c_2_elevated.effectiveDateTime_start, '-1 day')
+  )
+  JOIN pt ON pt.PatientId = json_extract(o.json, '$.subject.reference_id_aa')
+  AND DATETIME(
+    json_extract(o.json, '$.effectiveDateTime_aa.start')
+  ) >= DATETIME(pt.birthDate, '+18 years')
+  join json_each(json_extract(o.json, '$.code.coding')) observation_coding on (
+    json_extract(observation_coding.value, '$.system') = 'http://loinc.org'
```

Comment:
don't be fooled by ".value" here -- that's not a FHIR property but rather a structure created by the table-valued `json_each` function.